src/GameLogic/NPC/Monster.cs

```diff
@@ -227,23 +227,38 @@ internal async ValueTask RandomMoveAsync()
         {
             return;
         }
+        
+        byte randx = (byte)GameLogic.Rand.NextInt(Math.Max(0, this.Position.X - 1), Math.Min(0xFF, this.Position.X + this.Definition.MoveRange));
```

Comment:
The 2 is correct. It's randomly moving by one step when it has no target and is watched by a player.
I'm note sure what the MoveRange actually means.
If you change it like that, the monsters will all walk towards max x and y

Reply:
Rand(0, max) - sounds good?

Reply:
Okay, but when you add MoveRange just to the second parameter of `Rand.NextInt`, the character will more likely move to a x coordinate which is higher than the current. How about:
```
var moveByMaxX = GameLogic.Rand.NextInt(1, this.Definition.MoveRange + 1);
var moveByMaxY = GameLogic.Rand.NextInt(1, this.Definition.MoveRange + 1);
byte randx = (byte)GameLogic.Rand.NextInt(Math.Max(0, this.Position.X - moveByMaxX), Math.Min(0xFF, this.Position.X + moveByMaxX + 1));
byte randy = (byte)GameLogic.Rand.NextInt(Math.Max(0, this.Position.Y - moveByMaxY), Math.Min(0xFF, this.Position.Y + moveByMaxY + 1));
```

Reply:
added